src/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixer.php

```diff
@@ -61,40 +62,46 @@ public function isCandidate(Tokens $tokens)
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $functionyTokens = $this->getFunctionyTokenKinds();
-        $languageConstructionTokens = $this->getLanguageConstructionTokenKinds();
-        $braceTypes = $this->getBraceAfterVariableKinds();
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $functionyTokenKinds = $this->getFunctionyTokenKinds();
+        $specialConstructTokenKinds = $this->getSpecialConstructTokenKinds();
+        $braceTokens = $this->getBraceAfterVariableTokens();
 
         foreach ($tokens as $index => $token) {
-            // looking for start brace
+            // looking for opening parenthesis
             if (!$token->equals('(')) {
                 continue;
             }
 
-            // last non-whitespace token, can never be `null` always at least PHP open tag before it
-            $lastTokenIndex = $tokens->getPrevNonWhitespace($index);
+            // previous non-whitespace token, can never be `null` always at least PHP open tag before it
+            $prevNonWhitespaceIndex = $tokens->getPrevNonWhitespace($index);
 
-            // check for ternary operator
-            $endParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
-            $nextNonWhiteSpace = $tokens->getNextMeaningfulToken($endParenthesisIndex);
+            // check for special construct with binary or ternary operator (or comma for echo)
+            $closingParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+            $nextMeaningfulIndex = $tokens->getNextMeaningfulToken($closingParenthesisIndex);
             if (
-                null !== $nextNonWhiteSpace
-                && $tokens[$nextNonWhiteSpace]->equals('?')
-                && $tokens[$lastTokenIndex]->isGivenKind($languageConstructionTokens)
+                null !== $nextMeaningfulIndex
+                && $tokens[$prevNonWhitespaceIndex]->isGivenKind($specialConstructTokenKinds)
+                && (
+                    $tokens[$nextMeaningfulIndex]->isGivenKind(T_INSTANCEOF)
+                    || $tokens[$nextMeaningfulIndex]->equals('?')
+                    || ($tokens[$prevNonWhitespaceIndex]->isGivenKind(T_ECHO) && $tokens[$nextMeaningfulIndex]->equals(','))
+                    || (!$tokens[$nextMeaningfulIndex]->isGivenKind(T_DOUBLE_ARROW) && $tokensAnalyzer->isBinaryOperator($nextMeaningfulIndex))
```

Comment:
Wait, that's actually still not enough, I have discovered in https://github.com/php/php-src/blob/PHP-7.4/Zend/zend_language_parser.y that `print` is actually an operator (although currently not documented in https://www.php.net/operators.precedence) with _higher_ precedence than `and` (and `xor` and `or`), so `print $x and f()` is evaluated as `(print $x) and f()`, not as `print ($x and f())`, and so this PR shouldn't change the current behavior which fixes that `print ($x) and f()` to `print($x) and f()`... I will amend the code in the next few days